SPEC.md

```diff
@@ -0,0 +1,28 @@
+# Navigation.js specification
+
+## Purpose
+The goal of this specification is to outline the rules and ideas this library adheres to guide a user along a route.
+
+### Units
+By default, the metric system should be used where available.
+
+### Concepts
+Navigation.js looks at a single step along a greater route. All calculations are done against this step and not against the whole route. Once a step is completed, the next step on the route can be compared. If the user is no longer on the current step, the developer should re-route the user and also set their internal step counter back to 0.
+
+### Re-routing
+If at any time the user is not within `45 meters` of any point on the current step, re-route the user.
```

Comment:
Additionally, we need to explain where the numbers are coming from. This will help cases like the one @1ec5 describes below (notifications in 1 miles vs 1 km).